components/shapediver/parameter/ParameterSelectionComponent.tsx

```diff
@@ -0,0 +1,110 @@
+
+	// get the parameter API as we might have to overwrite the acceptRejectMode
+	const sessionApi = useShapeDiverStoreViewer(state => { return state.sessions[props.sessionId]; });
+	const parameterApi = sessionApi?.getParameterByName(props.parameterId)[0];
+
+	let acceptRejectMode = props.acceptRejectMode;
+	if(parameterApi && isInteractionSelectionParameterSettings(parameterApi.settings as any)) {
+		const selectionProps = parameterApi.settings?.props as ISelectionParameterSettings;
+		if((selectionProps?.minimumSelection ?? 1) < (selectionProps?.maximumSelection ?? 1)) {
+			acceptRejectMode = true;
+		}
+	}
+
+	const {
+		definition,
+		handleChange,
+		onCancel,
+		disabled,
+		value,
+		state
+	} = useParameterComponentCommons<string>({
+		acceptRejectMode,
+		disableIfDirty: props.disableIfDirty,
+		sessionId: props.sessionId,
+		parameterId: props.parameterId,
+		overrides: props.overrides,
+	}, 0);
+	
+	// state for the selection application
+	const [selectionActive, setSelectionActive] = useState<boolean>(false);
+	const { selectedNodes, resetSelectedNodeNames } = useSelection(props.sessionId, VIEWPORT_ID, selectionActive ? definition.settings as IInteractionParameterSettings : undefined);
+
+	useEffect(() => {
+		const parameterValue: SelectionParameterValue = { names: selectedNodes };
+		handleChange(JSON.stringify(parameterValue));
+	}, [selectedNodes]);
+
+	useEffect(() => {
+		if(acceptRejectMode === true && state.execValue === value) {
+			setSelectionActive(false);
+			resetSelectedNodeNames();
+		}
+	}, [state.execValue]);
+
+	/**
+	 * Callback function for the cancel button.
+	 */
+	const onCancelCallback = useCallback(() => {
+		setSelectionActive(false);
+		resetSelectedNodeNames();
+		
+		if(onCancel)
+			onCancel();
+	}, [selectionActive, resetSelectedNodeNames]);
+
+	const iconFinger = <Icon type={IconTypeEnum.IconHandFinger} />;
+	const iconInfo = <Icon type={IconTypeEnum.IconInfoCircleFilled} />;
+	const contentActive =
+		<Group style={{ flexDirection: "column" }} align="left">
+			<Text style={{ textAlign: "left" }} size="sm" fw={500}>
+				Select your object
+			</Text>
+			<Group>
+				<Text size="sm" fw={400} fs="italic">
+					Click on an object to select it in the viewer
+				</Text>
+			</Group>
+		</Group>;
+	const contentInactive =
+		<Text size="sm" fw={500}>
+			Select your object
+		</Text>;
+
+	return <>
+		<Group>
+			<ParameterLabelComponent {...props} cancel={selectionActive ? onCancel && onCancelCallback : undefined} />
+			{
+				definition &&
+				<Button justify="space-between" fullWidth h="100%" disabled={disabled}
+					rightSection={selectionActive ? iconInfo : iconFinger}
+					bg={selectionActive ? "" : "blue"}
+					onClick={() => setSelectionActive(!selectionActive)}>
+					<Group justify="space-between" w="100%" pt={"sm"} pb={"sm"}>
+						{selectionActive ? contentActive : contentInactive}
+					</Group>
+				</Button>
+			}
+		</Group>
```

Comment:
@MajorMeerkatThe3rd a general comment about using styling properties: Let's keep them as minimal as possible, for example try to avoid font weight, explicit definition of padding, etc. Keeping this minimal ensures there will be no problems with the theme (we want things to look good even if someone changes padding globally via the theme). 
In case specifying specific styles can't be avoided, there is also the possibility to get theme properties (like default padding) and use them, or define theme properties for the component. There are lots of examples on how to do that in other components. 
@mathieu-shapediver FYI